geode-core/src/main/java/org/apache/geode/redis/internal/executor/list/ListExecutor.java

```diff
@@ -71,7 +71,7 @@ protected void pushElements(ByteArrayWrapper key, List<byte[]> commandElems, int
     String oppositeKey = pushType == ListDirection.RIGHT ? "head" : "tail";
     Integer index = (Integer) keyRegion.get(indexKey);
     Integer opp = (Integer) keyRegion.get(oppositeKey);
-    if (index != opp)
+    if (!index.equals(opp))
```

Comment:
The other option here is to change the Integers to ints.

Reply:
may hit an NPE if gets return null